peps/pep-0776.rst

```diff
@@ -0,0 +1,792 @@
+PEP: 776
+Title: Emscripten Support
+Author: Hood Chatham <roberthoodchatham at gmail.com>
+Sponsor: Łukasz Langa <lukasz at python.org>
+Discussions-To: https://discuss.python.org/t/pep-776-emscripten-support/84996
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 18-Mar-2025
+Python-Version: 3.14
+Post-History: `18-Mar-2025 <https://discuss.python.org/t/pep-776-emscripten-support/84996>`__
+
+========
+Abstract
+========
+
+`Emscripten <https://emscripten.org/>`__ is a complete open source compiler
+toolchain which compiles C/C++ code into WebAssembly/JavaScript executables that
+run in JavaScript runtimes including browsers and Node.js.
+
+This PEP formalizes the addition of Tier 3 for Emscripten support in Python 3.14
+which `was approved by the steering council on October 25, 2024
+<https://github.com/python/steering-council/issues/256>`__. The goal is to allow
+Pyodide wheels to be uploaded to PyPI.
+
+==========
+Motivation
+==========
+
+A web browser is a universal computing platform, available on Windows, macOS,
+Linux, and every smartphone.
+
+`The Pyodide project <https://pyodide.org/>`__ has for years supported
+Emscripten Python. Hundreds of thousands of students have learned Python through
+Pyodide via projects like `Capytale
+<https://web.archive.org/web/20241211090946/https://cfp.jupytercon.com/2023/talk/TJ9YEV/>`__
+and `PyodideU <https://stanford.edu/~cpiech/bio/papers/pyodideU.pdf>`__. Pyodide
+is also increasingly being used by Python packages to provide interactive
+documentation. This demonstrates both the importance and the maturity of the
+Emscripten platform.
+
+Emscripten and WASI are also the only supported platforms that offer any
+meaningful sandboxing.
+
+=====
+Goals
+=====
```

Comment:
In general we try and stick to the suggested sections in [PEP 12](https://peps.python.org/pep-0012/#suggested-sections), perhaps this could be 'Rationale' (or even part of the specification).

Also, please could you change to only underlined headings? You can use tildes or carets if you need more nesting than simply sections and subsections.

Reply:
> Also, please could you change to only underlined heading?

I have four nested levels of headings. The overlined headings are suggested here:
https://peps.python.org/pep-0012/#section-headings

I am happy to switch to whatever convention is preferred but perhaps pep 12 should be updated if people don't like the overlined headings. I agree that they are a bit confusing.

Reply:
Okay I moved back to the headings I originally used instead of the PEP 12 headings.

Reply:
> In general we try and stick to the suggested sections in [PEP 12](https://peps.python.org/pep-0012/#suggested-sections)

I think "Goals" makes more sense as a top level section than a subsection or "Rationale". The "Suggested Settings" section of PEP 12 makes no statement to the effect that other section names is not allowed. So I think I'll keep it this way unless someone else seconds the suggestion.